Header header
string name
location[] points